ethcore/account-state/src/state.rs

```diff
@@ -759,20 +759,21 @@ impl<B: Backend> State<B> {
 	}
 
 	/// Remove any touched empty or dust accounts.
-	pub fn kill_garbage(&mut self, touched: &HashSet<Address>, remove_empty_touched: bool, min_balance: &Option<U256>, kill_contracts: bool) -> TrieResult<()> {
-		let to_kill: HashSet<_> = {
-			self.cache.borrow().iter().filter_map(|(address, ref entry)|
-				if touched.contains(address) && // Check all touched accounts
-					((remove_empty_touched && entry.exists_and_is_null()) // Remove all empty touched accounts.
+	pub fn kill_garbage(&mut self, touched: &HashSet<Address>, min_balance: &Option<U256>, kill_contracts: bool) -> TrieResult<()> {
+		let to_kill: HashSet<_> =
+			touched.iter().filter_map(|address| { // Check all touched accounts
+				self.cache.borrow().get(address).and_then(|entry| {
```

Comment:
I collected some more stats on the size of the cache on current mainnet. On the blocks I saw the cache length is invariably higher than size of the `touched` collection so I think this change – looping over `touched` rather than over the cached items – is good.

Reply:
`touched` is mostly 1, sometimes it goes up to ~10, rarely to ~50. The cache length is between a few tens to a few hundreds.